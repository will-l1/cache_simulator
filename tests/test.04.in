1024
65536
10
22 
1034
16400
17424
32790
33814
49176
50200
64504
65528
stats
