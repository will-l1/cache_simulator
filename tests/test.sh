#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
