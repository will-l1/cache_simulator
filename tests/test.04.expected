Loaded value [0xb9cb17b29e5109d2] @ address 0x00000016
Loaded value [0x1f87145407a839f2] @ address 0x0000040a
Loaded value [0x4d147b600626c106] @ address 0x00004010
Loaded value [0x51f1992c39ecb033] @ address 0x00004410
Loaded value [0x22876436fae2133f] @ address 0x00008016
Loaded value [0xa53c60d4396438c3] @ address 0x00008416
Loaded value [0x40fbdea34dc564da] @ address 0x0000c018
Loaded value [0x0b089f7a30e8efdd] @ address 0x0000c418
Loaded value [0x78d80dcc0da40dfc] @ address 0x0000fbf8
Loaded value [0x46b4412250a28152] @ address 0x0000fff8
Cache hits: 0, misses: 10 -- hit rate 0%
